public/app/core/reducers/navModel.test.ts

```diff
@@ -38,4 +37,123 @@ describe('applicationReducer', () => {
+              url: '/org/users',
+            },
+            {
+              id: 'teams',
+              text: 'Teams',
+              description: 'Manage org groups',
+              icon: 'users-alt',
+              url: '/org/teams',
+            },
+            {
+              id: 'plugins',
+              text: 'Plugins',
+              description: 'View and configure plugins',
+              icon: 'plug',
+              url: '/plugins',
+            },
+            {
+              id: 'org-settings',
+              text: 'Preferences',
+              description: 'Organization preferences',
+              icon: 'sliders-v-alt',
+              url: '/org',
+            },
+            {
+              id: 'apikeys',
+              text: 'API Keys',
+              description: 'Create & manage API keys',
+              icon: 'key-skeleton-alt',
+              url: '/org/apikeys',
+            },
+          ],
+          icon: 'cog',
+          id: 'cfg',
+          sortWeight: -1400,
+          subTitle: 'Organization: Jake Org',
+          text: 'Configuration',
+          url: '/datasources',
+        },
+        datasources: {
+          id: 'datasources',
+          text: 'Data Sources',
+          description: 'Add and configure data sources',
+          icon: 'database',
+          url: '/datasources',
+        },
+        users: {
+          id: 'users',
+          text: 'Users',
+          description: 'Manage org members',
+          icon: 'user',
+          url: '/org/users',
+        },
+        teams: {
+          id: 'teams',
+          text: 'Teams',
+          description: 'Manage org groups',
+          icon: 'users-alt',
+          url: '/org/teams',
+        },
+        plugins: {
+          id: 'plugins',
+          text: 'Plugins',
+          description: 'View and configure plugins',
+          icon: 'plug',
+          url: '/plugins',
+        },
+        'org-settings': {
+          id: 'org-settings',
+          text: 'Preferences',
+          description: 'Organization preferences',
+          icon: 'sliders-v-alt',
+          url: '/org',
+        },
+        apikeys: {
+          id: 'apikeys',
+          text: 'API Keys',
+          description: 'Create & manage API keys',
+          icon: 'key-skeleton-alt',
+          url: '/org/apikeys',
+        },
+      };
+
+      const orgName = 'New Org Name';
+      const subTitle = `Organization: ${orgName}`;
+      const expectedState = {
+        ...initialState,
+        cfg: { ...initialState.cfg, subTitle },
+        datasources: getItemWithNewSubTitle(initialState.datasources, subTitle),
+        users: getItemWithNewSubTitle(initialState.users, subTitle),
+        teams: getItemWithNewSubTitle(initialState.teams, subTitle),
+        plugins: getItemWithNewSubTitle(initialState.plugins, subTitle),
+        'org-settings': getItemWithNewSubTitle(initialState['org-settings'], subTitle),
+        apikeys: getItemWithNewSubTitle(initialState.apikeys, subTitle),
```

Comment:
Significant work adding tests but I think the expect part of any test should be raw and never use parts of the code it tests. I mean you shouldn't use `getItemWithNewSubTitle` in this expect but create the expected state yourself. Does that make sense @KrammerJake

Reply:
That makes perfect sense. I'll remove the usage of `getItemWithNewSubTitle` in my next commit.

Reply:
this initial state seems to big for what we are testing, is all this needed?

Reply:
@torkelo I reduced the size of the initial state in my latest commit and only kept the necessities. Don't know what I was thinking. 😄